src/components/DonutChart.js

```diff
@@ -0,0 +1,234 @@
+      animateOnHover: true,
+      animateOnLoad: true,
+      arcWidth: 80,
+      baseArcColor: '#E5E5E5',
+      colors: d3.scale.category20().range(),
+      data: [],
+      dataPoints: [],
+      dataPointColors: d3.scale.category20b().range(),
+      dataPointRadius: 40,
+      height: 360,
+      labelStyle: {},
+      onClick () {},
+      onMouseEnter () {},
+      onMouseLeave () {},
+      opacity: 1,
+      padAngle: 0.01,
+      radiusOffset: 0,
+      showBaseArc: false,
+      showLegend: true,
+      valueFormat: {},
+      valueStyle: {},
+      width: 360,
+      zeroStateText: 'Roll over an item for details'
+    };
+  },
+
+  _renderArcs () {
+    if (this.props.data.length > 0) {
+      const dataSets = this.props.data.map(item => {
+        return item.value;
+      });
+
+      const valueTotal = dataSets.reduce((a,b) => {
+        return a + b;
+      });
+
+      const endAngle = this.props.chartTotal ? valueTotal / this.props.chartTotal : 1;
+
+      const pie = d3.layout.pie().sort(null).padAngle(this.props.padAngle).endAngle(endAngle * 2 * Math.PI);
+      const values = pie(dataSets);
+      const radius = (Math.min(this.props.width, this.props.height) / 2) - this.props.radiusOffset;
+      const standardArc = d3.svg.arc().outerRadius(radius - this.props.activeOffset).innerRadius(radius - this.props.arcWidth);
+      const hoverArc = d3.svg.arc().outerRadius(radius).innerRadius(radius - this.props.arcWidth);
+
+
+      return values.map((point, i) => {
+        const arc = this.props.activeIndex === i && this.props.animateOnHover ? hoverArc : standardArc;
+
+        return (
+          <g
+            key={i}
+            onClick={this.props.onClick.bind(null, i)}
+            onMouseEnter={this.props.onMouseEnter.bind(null, i)}
+            onMouseLeave={this.props.onMouseLeave}
+          >
+            <DonutPath {...this.props} arc={arc} color={this.props.colors[i]} value={point} />
+          </g>
+        );
+      });
+    }
+  },
+
+  _renderBaseArc () {
+    if (this.props.showBaseArc) {
+      const radius = (Math.min(this.props.width, this.props.height) / 2) - this.props.radiusOffset;
+      const baseArc = d3.svg.arc().outerRadius(radius - this.props.activeOffset)
+                                  .innerRadius(radius - this.props.arcWidth)
+                                  .startAngle(0)
+                                  .endAngle(2 * Math.PI);
+
+      return (
+        <g
+          key={baseArc}
+        >
+          <path d={baseArc()} fill={this.props.baseArcColor}></path>
+        </g>
+      );
+    }
+  },
+
+  _renderDataPoints () {
+    const dataPoints = this.props.dataPoints.map(dataPoint => {
+      return dataPoint.value;
+    });
+
+    const radius = (Math.min(this.props.width, this.props.height) / 2) - this.props.radiusOffset;
+
+    return dataPoints.map((dataPoint, index) => {
+      const endAngle = this.props.chartTotal ? dataPoint / this.props.chartTotal : dataPoint / 100;
```

Comment:
Instead of having a conditional here, could we set the default value of `this.props.chartTotal` to `100` in `getDefaultProps`?